lib/azure/signature.rb

```diff
@@ -237,10 +236,13 @@ def generate_string(verb, headers, auth_type)
     #
     def canonicalize_resource(uri)
       resource = '/' + account_name + (uri.path.empty? ? '/' : uri.path)
-      params = CGI.parse(uri.query.to_s).map { |k,v| [k.downcase, v] }
-      params.sort_by! { |k,v| k }
-      params.map! { |k,v| '%s:%s' % [k, v.map(&:strip).sort.join(',')] }
-      [resource, *params].join("\n")
+
+      nested = (uri.query_values(Array) || []).sort
```

Comment:
Given a url like this:

    http://myaccount.blob.core.windows.net/mycontainer?restype=container&comp=list&include=snapshots&include=metadata&include=uncommittedblobs

The 'nested' variable ends up looking like this:

    [["comp", "list"], ["include", "metadata"], ["include", "snapshots"], ["include", "uncommittedblobs"], ["restype", "container"]]

The catch is that you can have multiple, identical keys with different values, as you can see here.